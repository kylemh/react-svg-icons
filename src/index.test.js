import React from 'react';
import renderer from 'react-test-renderer';

/*
 * We don't want to import the src modules because Jest doesn't have a way to handle SVGs.
 * In this system, we're just rendering React components which happen to render `svgs`. This will
 * more closely emulate the manner with which our consumers will use our components, also.
 */
import * as ProductionBundleIconsMap from '../dist/index.production.esm';

const createSnapshotTest = (Component) => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};

const iconNames = Object.keys(ProductionBundleIconsMap);
const icons = Object.values(ProductionBundleIconsMap);

describe('svg-icons', () => {
  icons.forEach((Icon, index) => {
    describe(`${iconNames[index]}`, () => {
      it('renders', () => {
        createSnapshotTest(<Icon />);
      });
    });
  });
});
