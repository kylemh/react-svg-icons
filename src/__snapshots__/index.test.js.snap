// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nike/svg-icons Alert renders 1`] = `
<svg
  fill="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Exclamation point in a circle
  </title>
  <path
    d="M12 0C9.813 0 7.8.533 5.96 1.6A11.793 11.793 0 0 0 1.6 5.96C.533 7.8 0 9.813 0 12s.533 4.2 1.6 6.04a11.793 11.793 0 0 0 4.36 4.36C7.8 23.467 9.813 24 12 24s4.2-.533 6.04-1.6a11.793 11.793 0 0 0 4.36-4.36C23.467 16.2 24 14.187 24 12s-.533-4.2-1.6-6.04a11.793 11.793 0 0 0-4.36-4.36C16.2.533 14.187 0 12 0zm0 17.36a1.31 1.31 0 0 1-.96-.4c-.267-.267-.4-.6-.4-1s.133-.733.4-1 .587-.4.96-.4.693.133.96.4.4.6.4 1-.133.733-.4 1a1.31 1.31 0 0 1-.96.4zm1.44-4.72h-2.8V6h2.8v6.64z"
  />
</svg>
`;

exports[`@nike/svg-icons Flag renders 1`] = `
<svg
  fill="currentColor"
  viewBox="0 0 19 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Flag
  </title>
  <path
    d="M18.96 4v12.64H2.56V28H0V4h18.96z"
  />
</svg>
`;

exports[`@nike/svg-icons Home renders 1`] = `
<svg
  fill="currentColor"
  viewBox="0 0 29 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Home
  </title>
  <path
    d="M23.44 13.36l-4-4.08L14.32 4 0 18.56h3.44V28H12v-6.88h4.56V28h8.56v-9.44h3.44z"
  />
</svg>
`;

exports[`@nike/svg-icons MagnifyingGlass renders 1`] = `
<svg
  fill="currentColor"
  viewBox="0 0 25 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Magnifying glass
  </title>
  <path
    d="M19.04 20.4q1.84-2.72 1.84-5.92 0-2.88-1.4-5.28t-3.8-3.8T10.44 4 5.2 5.4 1.4 9.2 0 14.44t1.4 5.24 3.8 3.8 5.28 1.4q3.76 0 6.72-2.48l5.52 5.6 2-2zm-8.56 1.68q-2.08 0-3.84-1.04t-2.8-2.8-1.04-3.8 1.04-3.8 2.8-2.8 3.8-1.04 3.8 1.04 2.8 2.8 1.04 3.8-1.04 3.8-2.8 2.8-3.76 1.04z"
  />
</svg>
`;
